std/algorithm/comparison.d

```diff
@@ -1567,3 +1567,156 @@ unittest
                 2, "two",
                 ));
 }
+
+/** Returns: First Argument (element of $(D a)) whose implicit conversion to
+    $(D bool) is $(D true.
+
+    Similar to behaviour of or operator in dynamic languages such as Lisp's (or
+    a...) and Python's a or ....
+
+    TODO: Is inout Conversion!T the correct return value?
+
+    NOTE: Lazy parameters are currently, too restrictively, inferred by DMD to
+    always throw eventhough they don't need to be. This makes it impossible to
+    mark $(D either) as $(D nothrow). Issue at
+    https://issues.dlang.org/show_bug.cgi?id=12647
+*/
+CommonType!Ts either(Ts...)(lazy Ts a)
+if (a.length >= 1 && !is(CommonType!Ts == void))
```

Comment:
Better also check that everything can actually convert to bool.

    enum implicitlyConvertsToBool(T) = isImplicitlyConvertible!(T, bool);
    allSatisfy(implicitlyConvertsToBool, Ts);